### DISCUSSION TITLE
Consider not tying formats so heavily to DXGI/D3D

### DISCUSSION THREAD

**Discussion by john-h-k** [SixLabors/ImageSharp.Textures#28] [Ideas]
Currently the formats seem to be directly tied to the `DXGI_FORMAT` enum from `dxgi.h`. Is this a conscious design decision or will it change? Ideally, I am planning to use this library for texture loading for both DirectX and Vulkan, and so a more platform agnostic system would be more useful, as well as allowing support for formats that aren't supported by `DXGI_FORMAT` (such as `double` precision pixels)

**Comment by JimBobSquarePants** [comment]
Pixel formats are extensible but I agree, if we can add some well known formats supported by Vulkan etc then that would be useful.

**Comment by brianpopow** [comment]
I have added support for decoding ktx textures along with support for more pixel formats (still its not exhaustive, there are just too many in opengl).

@john-h-k could you be more specific in which pixel formats you be interested in?

Also some compressions are not yet supported, like astc and etc. Any help in that regard would be very much appreciated.
